app/api/suggestions/suggestions.ts

```diff
@@ -0,0 +1,164 @@
+      [
+        { $match: { ...filters, status: 'ready' } },
+        {
+          $lookup: {
+            from: 'entities',
+            let: {
+              localField: '$entityId',
+            },
+            pipeline: [
+              {
+                $match: {
+                  $expr: {
+                    $and: [
+                      { $eq: ['$sharedId', '$$localField'] },
+                      { $eq: ['$language', language] },
+                    ],
+                  },
+                },
+              },
+            ],
+            as: 'entity',
+          },
+        },
+        {
+          $addFields: { entity: { $arrayElemAt: ['$entity', 0] } },
+        },
+        {
+          $addFields: {
+            currentValue: {
+              $cond: [
+                { $eq: ['$propertyName', 'title'] },
+                { v: [{ value: '$entity.title' }] },
+                {
+                  $arrayElemAt: [
+                    {
+                      $filter: {
+                        input: {
+                          $objectToArray: '$entity.metadata',
+                        },
+                        as: 'property',
+                        cond: {
+                          $eq: ['$$property.k', '$propertyName'],
+                        },
+                      },
+                    },
+                    0,
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        {
+          $addFields: {
+            currentValue: { $arrayElemAt: ['$currentValue.v', 0] },
+          },
+        },
+        {
+          $addFields: {
+            currentValue: { $ifNull: ['$currentValue.value', ''] },
+          },
+        },
+        {
+          $addFields: {
+            state: {
+              $switch: {
+                branches: [
+                  {
+                    case: { $eq: ['$suggestedValue', '$currentValue'] },
+                    then: 'Matching',
+                  },
+                  {
+                    case: {
+                      $eq: ['$currentValue', ''],
+                    },
+                    then: 'Empty',
+                  },
+                ],
+                default: 'Pending',
+              },
+            },
+          },
+        },
+        ...(state ? [{ $match: { $expr: { $eq: ['$state', state] } } }] : []),
+        { $sort: { date: 1, state: -1 } },
+        {
+          $project: {
+            entityId: '$entity._id',
+            sharedId: '$entity.sharedId',
+            entityTitle: '$entity.title',
+            language: 1,
+            propertyName: 1,
+            suggestedValue: 1,
+            segment: 1,
+            currentValue: 1,
+            state: 1,
+            page: 1,
+          },
+        },
+      ],
+      {
+        stage1: [{ $group: { _id: null, count: { $sum: 1 } } }],
+        stage2: [{ $skip: offset }, { $limit: limit }],
```

Comment:
I don't have a better solution at the moment, but this request could become a potential performance bottleneck for a large amount of entities (i.e. it needs to go through all the data for every page).

Reply:
Probably it could be, we need to do some tests with a large number of entities, even for the training phase.  Since it's against the database directly the indexes should help in order to speed the query, probably we'll need additional ones depending on the load tests.

Reply:
Agreed. Let's handle this as a tech debt, since I don't think it should hold up the rest of the pr, if you agree. I have meant this more as a conversation opener.